### DISCUSSION TITLE
Question about "Initial consumer offset"

### DISCUSSION THREAD

**Discussion by yehaifeng** [apache/rocketmq#6309] [Q&A]
> Apache RocketMQ uses the maximum offset of a message queue when a consumer obtains messages from the queue for the first time as the initial consumer offset. In other words, the consumer starts consuming from the latest message in the queue.

Why is `maximum offset`, while not `minimum offset`

**Comment by socutes** [comment]
I think the earliest and the latest have their own disadvantages。
minimum offset: There will be lots of reconsumption.
maximum offset：There will be old data that will not be consumed
By default, neither is better than the other.
